component/model.go

```diff
@@ -554,3 +556,112 @@ func (c *ModelComponent) UploadFile(ctx context.Context, req *types.CreateFileRe
 
 	return err
 }
+
+func (c *ModelComponent) SDKListFiles(ctx context.Context, namespace, name string) (*types.SDKFiles, error) {
+	var sdkFiles []types.SDKFile
+	model, err := c.ms.FindByPath(ctx, namespace, name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find model, error: %w", err)
+	}
+	filePaths, err := getFilePaths(namespace, name, "", c.gs.GetModelFileTree)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get all model files, error: %w", err)
+	}
+
+	for _, filePath := range filePaths {
+		sdkFiles = append(sdkFiles, types.SDKFile{Filename: filePath})
+	}
+	return &types.SDKFiles{
+		ID:        fmt.Sprintf("%s/%s", namespace, name),
+		Siblings:  sdkFiles,
+		Private:   model.Private,
+		Downloads: model.Downloads,
+		Likes:     model.Likes,
+		Tags:      []string{},
+		SHA:       model.Repository.DefaultBranch,
+	}, nil
+}
+
+func getFilePaths(namespace, repoName, folder string, gsTree func(namespce, repoName, ref, path string) ([]*types.File, error)) ([]string, error) {
+	var filePaths []string
+	gitFiles, err := gsTree(namespace, repoName, "", folder)
+	if err != nil {
+		slog.Error("Failed to get dataset file contents", slog.String("path", folder), slog.Any("error", err))
+		return filePaths, err
+	}
+	for _, file := range gitFiles {
+		if file.Type == "dir" {
+			subFileNames, err := getFilePaths(namespace, repoName, file.Path, gsTree)
+			if err != nil {
+				return filePaths, err
+			}
+			filePaths = append(filePaths, subFileNames...)
+		} else {
+			filePaths = append(filePaths, file.Path)
+		}
+	}
+	return filePaths, nil
+}
+
+func (c *ModelComponent) IsLfs(ctx context.Context, req *types.GetFileReq) (bool, error) {
+	content, err := c.gs.GetModelFileRaw(req.Namespace, req.Name, req.Ref, req.Path)
+
+	if err != nil {
+		slog.Error("failed to get model file raw", slog.String("namespace", req.Namespace), slog.String("name", req.Name), slog.String("path", req.Path))
+		return false, err
+	}
+
+	return strings.HasPrefix(content, LFSPrefix), nil
+}
+
+func (c *ModelComponent) HeadDownloadFile(ctx context.Context, req *types.GetFileReq) (*types.File, error) {
```

Comment:
why not reuse method DownloadFile? what's the difference?

Reply:
The difference between `DownloadFile` and `SDKDownloadFile` is the `SDKDownloadFile` will automatically determine if the file to download is an lfs file and decide which logic to use.